apps/src/lib/levelbuilder/code-docs-editor/ProgrammingClassEditor.jsx

```diff
@@ -0,0 +1,221 @@
+  };
+
+  return (
+    <div>
+      <h1>{`Editing Class "${key}"`}</h1>
+      <h2>
+        This feature is in development. Please continue to use curriculum
+        builder to edit code documentation.
+      </h2>
+      <label>
+        Display Name
+        <input
+          value={programmingClass.name}
+          onChange={e => updateProgrammingClass('name', e.target.value)}
+          style={styles.textInput}
+        />
+      </label>
+      <label>
+        Key (Used in URLs)
+        <input value={key} readOnly style={styles.textInput} />
+      </label>
+      <label>
+        Category
+        <select
+          value={programmingClass.categoryKey}
+          onChange={e => updateProgrammingClass('categoryKey', e.target.value)}
+          style={styles.selectInput}
+        >
+          <option key="none" value={''}>
+            (None)
+          </option>
+          {environmentCategories.map(category => (
+            <option key={category.key} value={category.key}>
+              {category.name}
+            </option>
+          ))}
+        </select>
+        <HelpTip>
+          Choose a category for the code documentation to fall beneath
+        </HelpTip>
+      </label>
+      <CollapsibleEditorSection title="Documentation" collapsed>
+        <label>
+          External Documentation
+          <HelpTip>Link to external documentation</HelpTip>
+          <input
+            value={programmingClass.externalDocumentation}
+            onChange={e =>
+              updateProgrammingClass('externalDocumentation', e.target.value)
+            }
+            style={styles.textInput}
+          />
+        </label>
+        <TextareaWithMarkdownPreview
+          markdown={programmingClass.content}
+          label={'Content'}
+          handleMarkdownChange={e =>
+            updateProgrammingClass('content', e.target.value)
+          }
+          features={markdownEditorFeatures}
+        />
+      </CollapsibleEditorSection>
+      <CollapsibleEditorSection title="Details" collapsed>
+        <TextareaWithMarkdownPreview
+          markdown={programmingClass.syntax}
+          label={'Syntax'}
+          handleMarkdownChange={e =>
+            updateProgrammingClass('syntax', e.target.value)
+          }
+          features={markdownEditorFeatures}
+        />
+      </CollapsibleEditorSection>
+      <CollapsibleEditorSection title="Tips" collapsed>
+        <TextareaWithMarkdownPreview
+          markdown={programmingClass.tips}
+          label={'Tips'}
+          handleMarkdownChange={e =>
+            updateProgrammingClass('tips', e.target.value)
+          }
+          features={markdownEditorFeatures}
+          helpTip="List of tips for using this code documentation"
+        />
+      </CollapsibleEditorSection>
+      <CollapsibleEditorSection title="Examples" collapsed>
+        <OrderableList
+          list={programmingClass.examples || []}
+          setList={list => updateProgrammingClass('examples', list)}
```

Comment:
I see unique keys are set when the component mounts, do we need to update the key when the example changes?

Reply:
Good question! This is actually handled by [`OrderableList`](https://github.com/code-dot-org/code-dot-org/blob/9a6774a23f69939665d790558d259db96abfad78/apps/src/lib/levelbuilder/code-docs-editor/OrderableList.jsx#L14), a component I've been using for anything that should be reordered. Handling the keys in general has been a bit tricky but in the case of examples, they don't have a natural unique identifier from the server (they're stored as a json blob), so this works to keep track of them through editing